dsl/camel-kamelet-main/src/main/java/org/apache/camel/main/download/KnownDependenciesResolver.java

```diff
@@ -80,4 +80,13 @@ public MavenGav mavenGavForClass(String className) {
         }
         return answer;
     }
+
+    private String findGav(String prefix) {
+        String gav = mappings.get(prefix);
+        while (gav == null && prefix.lastIndexOf(".") != -1) {
```

Comment:
You cannot keep dropping sub package names, which can end up into picking the wrong JAR. 
It should only drop the classname

Reply:
But the class may be in a subpackage, e.g. consider com.example:somejar:1.0.0 and two classes com.example.somejar.subpackage.SomeClass
com.example.somejar.otherpackage.OtherClass

Reply:
Hmm its an use-case outside camel-jbang, but it may not cause trouble. But mind if anything is pain then its removed.

The unit test files need to have ASF license headers. And also the properties file.

Reply:
I added the missing headers. Pls review